### DISCUSSION TITLE
typecasting number to char and Assert make the code not run.

### DISCUSSION THREAD

**Discussion by supawichza40** [nunit/nunit3-vs-adapter#996] [Q&A]
NUnit frame work use 3.13.3
NUnit3TestAdapter iinstalled 4.2.1

Hi, guys, I know that this is not a high priority, but I think it is a bug and must be reported, so I tried to compare a type casting of the number to char, and compare it to another string, but as a result, make that test not runnable.
and since you can run the same test on the console runner, they said that it must be a problem with the adapter, could you inform me if this is a bug, or is there a configuration you need to do.
![Screenshot 2022-07-13 195649](https://user-images.githubusercontent.com/88581422/178810372-c195d105-b355-4dd5-b731-19d1ad6ba3cc.png)

**Comment by OsirisTerje** [comment]
This is part of the FQN issues.  So yes, it is a bug, but there are multiple similar to this one, see https://github.com/nunit/nunit3-vs-adapter/issues?q=is%3Aopen+is%3Aissue+label%3Ais%3AFQNIssue  
We're waiting for a fix from the MS PG for these. 

@CharliePoole  @supawichza40 says it works with the console, but from the xml dump file I see that there is no events coming from the engine for the test.  It says it finds one test, so why is this working with the console and not the adapter/engine ?  I have enclosed the execution dump file. 
[E_Issue996.dll.zip](https://github.com/nunit/nunit3-vs-adapter/files/9112474/E_Issue996.dll.zip)

**Reply by CharliePoole** [reply]
I can try to set this up and run it under the console runner. One question... where are &quot; and &apos; coming from in the filter? If it's just an artifact of your dump, no problem. But the framework has to receive the actual filter string in order for it to work.

**Reply by OsirisTerje** [reply]
Not sure what "and" you mean:  The filter I see is : 
```xml
"<filter><test>Issue996.Tests.Test1(&quot;1&quot;,&apos;\u0001&apos;)</test></filter>"
```

**Reply by OsirisTerje** [reply]
Btw, I have added a repro project here, with a runsettings file that includes the dump setting:  https://github.com/nunit/nunit3-vs-adapter.issues/tree/master/Issue996

**Reply by supawichza40** [reply]
@CharliePoole for the "and" I meant as a word when asking question, but nothing related to the filter, it just a test case when doing string '1' with a type casting char(1) to '1', just like in the picture, sorry for the confusion.

**Comment by CharliePoole** [comment]
A quick note for clarification... the engine generates start-run and test-run events. All other events are merely passed through from the framework. So if there is no test-suite or test-case event, that means the framework never ran the tests.

**Reply by OsirisTerje** [reply]
If I remove the filter, the test run.  So, this looks like some FQN issue - The filter statement doesn't seem to be accepted by the framework, but what is the difference from running with the Console.  Would be awesome if you can check that out.

**Reply by OsirisTerje** [reply]
Also, if you run the test with
```cmd
dotnet test

```
The test will run.  The dotnet test command will not add any filter whatsoever unless one is explicitly given.  The reason the filter is there with Visual Studio is that we no longer (from some version in 2019 I think, they changed the call),  know if all tests are run or a selection, so we therefore have to assume we are being called with a selection.

**Comment by CharliePoole** [comment]
@OsirisTerje I see that on line 17 of the dump. But on line 43 I see

" (At Execution)  <filter><test>Issue996.Tests.Test1(&quot;1&quot;,&apos;\u0001&apos;)</test></filter>"

That's what I was asking about

**Reply by CharliePoole** [reply]
Odd... why is GitHub calling me a maintainer of this project?

**Reply by OsirisTerje** [reply]
Because you're part of the Project Owners team, you and Rob, and that team is set as Admins on this project too. :-)

**Reply by OsirisTerje** [reply]
About line 17:  That is just information from VS that is also written to the dump file.  The dump file is a mix of what we get from nunit and what comes from VS.

**Reply by CharliePoole** [reply]
But line 43 was what I was asking about. It contains html entities, which would not be understood by NUnit. I edited the above comment so they are now visible.

BTW, is there any reason your dump can't be made a valid XML file so it can be viewed in the browser? I have to use gedit to view it.

**Reply by OsirisTerje** [reply]
Absolutely, that can be done.  I'm always using Notepad++ since it autorefreshes.

**Comment by CharliePoole** [comment]
IMO adding the owners team is a mistake. It doesn't give us the authority to do anything we can't already do as owners but it misleads people as to our role in your project.  I notice yours is not the only project that has done this, however.

**Reply by OsirisTerje** [reply]
I can remove it here.
